### PR TITLE
Auto import keys in transactional systems right after registration

### PR DESCRIPTION
**Pull request by jlausuch**
https://progress.opensuse.org/issues/164183#note-4

VR https://openqa.suse.de/tests/14968601

### CODE REVIEW

File: tests/console/suseconnect_scc.pm [hunk 1]
```diff
@@ -31,17 +30,25 @@ sub run {
...
     # There are sporadic failures due to the command timing out, so we increase the timeout
     # and make use of retries to overcome a possible sporadic network issue.
-    my $output = script_output_retry("$cmd", retry => $retries, delay => $delay, timeout => 180);
-    die($output) if ($output =~ m/error|timeout|problem retrieving/i);
-    process_reboot(trigger => 1) if is_transactional;
+    # script_output_retry is useless for `transactional-update` cmd because it returns 0 even with failure
+    # trup_call will raise a failure if the command fails
+    if (is_transactional) {
+        trup_call('register' . $cmd);
+        trup_call('--continue run zypper --gpg-auto-import-keys refresh');
```
Comment (Vogtinator) [inline tests/console/suseconnect_scc.pm:43]: Only do `--gpg-auto-import-keys` if *really* necessary

Comment (jlausuch) [reply tests/console/suseconnect_scc.pm:43]: you mean with something like `is_staging` ?

Comment (Vogtinator) [reply tests/console/suseconnect_scc.pm:43]: Yes. Arguably it should also verify that it's the correct key somehow, but not sure how to get that info

**Comment by Vogtinator** [comment]
+1, though maybe we can get suse-build-key in stagings adjusted quick enough. Might get results very soon even.

File: tests/console/suseconnect_scc.pm (hunk 1 above)
Comment (jlausuch) [reply tests/console/suseconnect_scc.pm]: done

**Review by rfan1 — APPROVED** [review]
LTGM

**Comment by foursixnine** [comment]
> +1, though maybe we can get suse-build-key in stagings adjusted quick enough. Might get results very soon even.

I'd prefer this but I'm fine either way

**Review by foursixnine — APPROVED** [review]

**Pull request merged**